custom_components/octopus_energy/heat_pump/zone.py

```diff
@@ -89,42 +92,49 @@ def _handle_coordinator_update(self) -> None:
     """Retrieve the previous rate."""
 
     current = now()
-    result: HeatPumpCoordinatorResult = self.coordinator.data if self.coordinator is not None and self.coordinator.data is not None else None
+    result: HeatPumpCoordinatorResult = self.coordinator.data if self.coordinator is not None and self.coordinator.data is not None else None    
     if (result is not None and 
         result.data is not None and 
-        result.data.octoHeatPumpControllerConfiguration is not None and
-        result.data.octoHeatPumpControllerConfiguration.zones and
+        result.data.octoHeatPumpControllerStatus is not None and
+        result.data.octoHeatPumpControllerStatus.zones and
         (self._last_updated is None or self._last_updated < result.last_retrieved)):
       _LOGGER.debug(f"Updating OctopusEnergyHeatPumpZone for '{self._heat_pump_id}/{self._zone.configuration.code}'")
 
-      zones: List[ConfigurationZone] = result.data.octoHeatPumpControllerConfiguration.zones
+      zones: List[Zone] = result.data.octoHeatPumpControllerStatus.zones
       for zone in zones:
-        if zone.configuration is not None and zone.configuration.code == self._zone.configuration.code and zone.configuration.currentOperation is not None:
+        if zone.telemetry is not None and zone.zone == self._zone.configuration.code and zone.telemetry.mode is not None:
 
-          if zone.configuration.currentOperation.mode == "ON":
+          if zone.telemetry.mode == "ON":
             self._attr_hvac_mode = HVACMode.HEAT
             self._attr_preset_mode = PRESET_NONE
-          elif zone.configuration.currentOperation.mode == "OFF":
+          elif zone.telemetry.mode == "OFF":
             self._attr_hvac_mode = HVACMode.OFF
             self._attr_preset_mode = PRESET_NONE
-          elif zone.configuration.currentOperation.mode == "AUTO":
+          elif zone.telemetry.mode == "AUTO":
             self._attr_hvac_mode = HVACMode.AUTO
             self._attr_preset_mode = PRESET_NONE
-          elif zone.configuration.currentOperation.mode == "BOOST":
+          elif zone.telemetry.mode == "BOOST":
             self._attr_hvac_mode = HVACMode.AUTO
             self._attr_preset_mode = PRESET_BOOST
           else:
-            raise Exception(f"Unexpected heat pump mode detected: {zone.configuration.currentOperation.mode}")
+            raise Exception(f"Unexpected heat pump mode detected: {zone.telemetry.mode}")
 
-          self._attr_target_temperature = zone.configuration.currentOperation.setpointInCelsius
-          self._end_timestamp = datetime.fromisoformat(zone.configuration.currentOperation.end) if zone.configuration.currentOperation.end is not None else None
+          self._attr_hvac_action = HVACAction.HEATING if zone.telemetry.relaySwitchedOn else HVACAction.IDLE
```

Comment:
praise: Good to see this being added :) I was thinking it perhaps should use the current/target temperature to work this out (if current is less than target, then heating, otherwise idle), but I think this is more accurate.